server/protocols/protocols-library/src/main/java/org/apache/james/protocols/lib/netty/AbstractConfigurableAsyncServer.java

```diff
@@ -180,8 +182,14 @@ public final void configure(HierarchicalConfiguration<ImmutableNode> config) thr
         Integer bossWorker = config.getInteger("bossWorkerCount", null);
         setBossWorkerCount(Optional.ofNullable(bossWorker));
 
+        RejectedExecutionHandler noopRejectedHandler = (task, executor) -> {
+            // does not throw RejectedExecutionException upon executor shutdown
```

Comment:
throw if AbstractConfigurableAsyncServer is running (issue with too full executor backlog needs to be reported)

Do not throw once graceful shutdown is initialised.

We can track this in a variable of AbstractConfigurableAsyncServer